pilot/pkg/controller/workloadentry/workloadentry_controller.go

```diff
@@ -279,21 +283,31 @@ func (c *Controller) UpdateWorkloadEntryHealth(proxy *model.Proxy, event HealthE
 		return
 	}
 
-	// get previous status
-	cfg := c.store.Get(gvk.WorkloadEntry, entryName, proxy.Metadata.Namespace)
-	if cfg == nil {
-		log.Errorf("config was nil when getting WorkloadEntry %v for %v", entryName, proxy.ID)
-		return
-	}
+	condition := transformHealthEvent(event)
+	// backoff retry
+	// TODO(@hzxuzhonghu): handle race when reconnect happens and health status changed
+	for i := 0; i < maxRetries; i++ {
```

Comment:
This is going to block the entire ADS stream for the proxy. If we are going to do this might make more sense to just return an error?

Reply:
oh nevermind, I see you moved it to a goroutine